src/anemoi/training/diagnostics/callbacks/plot.py

```diff
@@ -112,27 +100,42 @@ def _output_figure(
             save_path = Path(
                 self.save_basedir,
                 "plots",
-                f"{tag}_epoch{epoch:03d}.png",
+                f"{tag}_epoch{epoch:03d}.jpg",
             )
 
             save_path.parent.mkdir(parents=True, exist_ok=True)
-            fig.savefig(save_path, dpi=100, bbox_inches="tight")
+
+            fig.canvas.draw()
+            image_array = np.array(fig.canvas.renderer.buffer_rgba())
+            plt.imsave(save_path, image_array, dpi=100)
```

Comment:
Can you explain these updates?
They seem seperate to the datashader updates.

Reply:
This is coming from https://github.com/ecmwf-lab/aifs-mono/pull/230. The aim of that PR was to bring some optimisations for the code to speed up the plotting. Mainly those are:
- Improved asynchronous callback using asyncio (better performance and can handle error catching)
- Introduce the option to use datashader as an alternative to Scatter plots
- Other optimisations like avoiding calling plt.tight_layout or speeding up the saving of plots. 
I can update the PR description with all this info so it helps understand the motivation behind all the changes